ProQuest.js

```diff
@@ -120,79 +120,96 @@ function fetchEmbeddedPdf(url, item, callback) {
+			break;
+		}
+	}
+	if (!root) {
+		var elements = doc.getElementsByClassName('resultListContainer');
+		for (var i=0; i<elements.length; i++) {
+			if (elements[i] && elements[i].offsetHeight>0) {
+				root = elements[i];
 				break;
 			}
 		}
-	} else {
-		root = doc.getElementsByClassName('resultListContainer')[0];
-		if (root && !root.offsetHeight) return false; // Not sure if this can actually happen
 	}
-
-	if (!root) return false;
+	if (!root) {
+		Z.debug("No root found");
+		return false;
+	}
 
 	var results = doc.getElementsByClassName('resultItem');
-	//ZU.xpath(root, './/a[contains(@class,"previewTitle") or contains(@class,"resultTitle")]');
-	
+
 	var items = {}, found = false;
 	for(var i=0, n=results.length; i<n; i++) {
-		var title = results[i].getElementsByClassName('resultTitle')[0]
-			|| results[i].getElementsByClassName('previewTitle')[0];
-		if (!title || title.nodeName != 'A') continue;
-		
-		if (checkOnly) return true;
-		found = true;
-		
-		items[title.href] = title.textContent;
+		if (results[i].offsetHeight>0) { //restrict to the active results only
+			var title = results[i].getElementsByClassName('resultTitle')[0]
+				|| results[i].getElementsByClassName('previewTitle')[0];
+			if (!title || title.nodeName != 'A') continue;
+			
+			if (checkOnly) return true;
+			found = true;
+			
+			items[title.href] = title.textContent;
+		}
 	}
 		
 	return found ? items : false;
 }
 
+var map = {
+	"newspaperArticle" : "newspaperArticle",
+	"jnlArticle" : "journalArticle", //also magazineArticles
+	"dissertationPublished" : "thesis",
+	"reportSerial" : "report",
+	"workingPaper" : "report",
+	"book" : "book",
+	"webPage" : "webpage",
+	"blog" : "blogPost",
+	"report" : "report",
+	"audioVideo" : "videoRecording",
+	"default" : "conferencePaper" //also default = manuscript?, referenceWork, encyclopedia
+}
+
 function detectWeb(doc, url) {
 	initLang(doc, url);
-
-	followLink = false;
-
+	
+	followLink = true;
+	var active = doc.getElementsByClassName('active');
+	for (var i=0; i<active.length; i++) {
+		if ( active[i].id.indexOf("tab-AbstractRecord-null") > -1) {
+			followLink = false;
+		}
+	}
+	
 	//Check for multiple first
-	if (url.indexOf('docview') == -1 &&
-		url.indexOf('pagepdf') == -1) {
-		if (getSearchResults(doc, true))
+	if (url.indexOf('docview') == -1 && url.indexOf('pagepdf') == -1) {
+		if (getSearchResults(doc, true)) {
 			return "multiple";
+		}
 	}
-
+	
+	//if we are on Abstract/Details page,
+	//then we can read the type from the corresponding field
 	var types = getTextValue(doc, ["Source type", "Document type", "Record type"]);
 	var zoteroType = getItemType(types);
-	if(zoteroType) return zoteroType;
-
-	//hack for NYTs, which misses crucial data.
-	var db = getTextValue(doc, "Database")[0];
-	if (db && db.indexOf("The New York Times") !== -1) {
-		return "newspaperArticle";
-	}
-
-	// Fall back on journalArticle-- even if we couldn't guess the type
-	if(types.length) return "journalArticle";
```

Comment:
Okay. Do you have examples that they are still needed? (AFAIK I don't have access to the NYT examples.)